File: src/frontend/src/interfaces/module.ts
import { IPaginationDataItem, IPaginationData } from './index';


export interface ITeamServerCreate {
    port: number;
    password: string;
    c2ProfileId: number;
    vpsId: number;
    killDate: string;
    remark: Nullable<string>;
    csDownloadUrl: string;
    zipPassword: string;
}

export interface ITeamServerItemData extends IPaginationDataItem {
    id: number;
    ispProviderName: string;
    hostname: string;
}

export interface ITeamServerPaginationData extends IPaginationData {
    items: ITeamServerItemData[];
}


export interface IRedirectorCreate {
    remark: Nullable<string>;
}

export interface IRedirectorItemData extends IPaginationDataItem {
    id: number;
    ispProviderName: string;
    templateSiteName: string;
}

export interface IRedirectorPaginationData extends IPaginationData {
    items: IRedirectorItemData[];
}

export interface IRedirectorCreateData {
    beaconType: string;
    teamServerId: number;
    listenerPort: number;
    redirectDomain: string;
    vpsId: number;
    domainId: number;
    remark: Nullable<string>;
}

File: src/backend/app/schemas/module.py
from typing import List
from typing import Optional

from app.models.module import RedirectorBeaconType
from app.schemas.base import BasePagination
from app.schemas.base import BaseSchema
from app.schemas.base import BaseSuccessfulResponseModel
from app.schemas.base import APIModel


class TeamServerCreateSchema(APIModel):
    port: int = 9990
    password: Optional[str]
    c2_profile_id: Optional[int]
    vps_id: int
    kill_date: Optional[str]
    remark: Optional[str] = None
    cs_download_url: str
    zip_password: Optional[str]


class TeamServerItemSchema(BaseSchema):
    port: Optional[int]
    password: Optional[str]
    c2_profile_id: Optional[int]
    vps_id: int
    kill_date: Optional[str]
    hostname: str
    ip: Optional[str]
    isp_provider_name: str
    c2_profile_name: Optional[str]


class TeamServerItemResponse(BaseSuccessfulResponseModel):
    result: TeamServerItemSchema


class TeamServerPaginationSchema(BasePagination):
    items: Optional[List[TeamServerItemSchema]]


class TeamServerPaginationResponse(BaseSuccessfulResponseModel):
    result: TeamServerPaginationSchema


class RedirectorCreateSchema(APIModel):
    beacon_type: RedirectorBeaconType
    team_server_id: int
    listener_port: int
    redirect_domain: str
    remark: Optional[str]
    vps_id: int
    domain_id: int


class RedirectorItemSchema(BaseSchema, RedirectorCreateSchema):
    beacon_type_name: str
    hostname: str
    domain_name: str
    ip: str


class RedirectorPaginationSchema(BasePagination):
    items: Optional[List[RedirectorItemSchema]]


class RedirectorPaginationResponse(BaseSuccessfulResponseModel):
    result: RedirectorPaginationSchema
